src/App.tsx

```diff
@@ -1,14 +1,41 @@
-import React from 'react';
+import React, { useState, useMemo, useCallback } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { Dropdown } from './components/Dropdown/Dropdown';
+import { debounce } from './helpers/debounce';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person>();
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+
+  const visablePeople = useMemo(() => {
+    return peopleFromServer.filter(person => person.name
+      .toLowerCase().includes(query.toLowerCase()));
+  }, [appliedQuery]);
```

Comment:
The useMemo dependency should be 'query' instead of 'appliedQuery'. The 'appliedQuery' state is never used and can be removed.